packages/@lwc/integration-karma/test/component/aria-reflection/index.spec.js

```diff
@@ -0,0 +1,119 @@
+import { createElement } from 'lwc';
+import { ariaPropertiesMapping, extractDataIds } from 'test-utils';
+import NoPropDeclared from 'x/noPropDeclared';
+import PropDeclared from 'x/propDeclared';
+import ApiPropDeclared from 'x/apiPropDeclared';
+
+describe('aria reflection', () => {
+    const scenarios = [
+        {
+            name: 'No prop declared',
+            tagName: 'x-no-prop-declared',
+            Ctor: NoPropDeclared,
+            expectAttrReflection: true,
+        },
+        {
+            name: 'Prop declared',
+            tagName: 'x-prop-declared',
+            Ctor: PropDeclared,
+            // declaring a prop in the component results in no attribute reflection
+            expectAttrReflection: false,
+        },
+        {
+            name: '@api prop declared',
+            tagName: 'x-api-prop-declared',
+            Ctor: ApiPropDeclared,
+            // declaring a prop in the component results in no attribute reflection
+            expectAttrReflection: false,
+        },
```

Comment:
Maybe we should add tests for `@track` too.

Reply:
Added a test, and confirmed that it worked the same in v3: bf9edb410